ambry-quota/src/main/java/com/github/ambry/quota/AmbryQuotaManager.java

```diff
@@ -141,4 +154,8 @@ public QuotaConfig getQuotaConfig() {
     }
     return quotaSourceObjectMap;
   }
+
+  public final static class AmbryQuotaManagerComponents {
+    public static AccountStatsStore accountStatsStore;
```

Comment:
I don't think there is really any need to create a separate AmbryQuotaManagerComponents class. Just make accountStatsStore private final and add a getter inside the AmbryQuotaManager itself.
Pass the accountStatsStore in AmbryQuotaManager's constructor.

Reply:
the problem is that when constructing a StorageQuotaAdaptorEnforcer, AmbryQuotaManager has not yet constructed. So we need a static accessor to set and get accountStatsStore.
One way to do this is to add static field in AmbryQuotaManager and set this field in the AmbryQuotaManagerFactory so we can get this field out in the StorageQuotaAdaptorEnforcer. But this way has no difference from having a static inner class.
Another possible solution is to pass accountStatsStore to all the enforcer factories' constructors, but I figure all the other enforcers don't need it, only storage enforcer need it, then it seems intrusive.